programs/bpf_loader/src/lib.rs

```diff
@@ -1598,17 +1596,11 @@ fn execute<'a, 'b: 'a>(
         }
         match result {
             ProgramResult::Ok(status) if status != SUCCESS => {
-                let error: InstructionError = if (status == MAX_ACCOUNTS_DATA_ALLOCATIONS_EXCEEDED
+                let error: InstructionError = if status == MAX_INSTRUCTION_TRACE_LENGTH_EXCEEDED
                     && !invoke_context
                         .feature_set
-                        .is_active(&cap_accounts_data_allocations_per_transaction::id()))
-                    || (status == MAX_INSTRUCTION_TRACE_LENGTH_EXCEEDED
```

Comment:
ideally don't combine feature gate logic with other logic to make these removal diffs cleaner